### PR TITLE
i#2876 clang-format: Add first version of clang-format style.

### PR DESCRIPTION
**Pull request by fhahn**
This patch adds a first version of a clang-format style definition for
DynamoRIO. I think it is relatively close to the style guide, but
please let me know of any other rules that should be accounted for.

I have re-formatted 3 files from the aarch64 backend. We do not have to
include them in the final commit, but I thought they would be useful to
illustrate the impact of the style rules.

Issue #2876

### CODE REVIEW

**Comment by fhahn** [comment]
Also, I do not think we should re-format everything, but rather use this to re-format new patches initially (aka ` git diff -U0 --no-color HEAD^ | clang-format-diff.py -i -p1`)

We can also add this to the CI, to check if the formatting is OK.

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+BasedOnStyle: WebKit
+AllowShortBlocksOnASingleLine: false
+AllowShortFunctionsOnASingleLine: false
+AllowShortIfStatementsOnASingleLine: false
+AlwaysBreakAfterDefinitionReturnType: All
```
Comment (derekbruening) [inline .clang-format]: We want AlwaysBreakAfterReturnType too which requires a recent clang-format.  I think we should ship a binary version for uniformity: it is changing too fast and we want too-recent features to rely on the distro package version.

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+BasedOnStyle: WebKit
+AllowShortBlocksOnASingleLine: false
+AllowShortFunctionsOnASingleLine: false
```
Comment (derekbruening) [inline .clang-format]: What version of clang-format are you using here?  See below where mine is no longer a boolean ("None").

File: .clang-format [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+AllowShortIfStatementsOnASingleLine: false
+AlwaysBreakAfterDefinitionReturnType: All
+AlignAfterOpenBracket: Align
+BinPackArguments: true
+BinPackParameters: true
+BreakBeforeBinaryOperators: None
+ColumnLimit: 100
```
Comment (derekbruening) [inline .clang-format]: 90, not 100

File: core/arch/aarch64/instr.c [hunk 4]
```diff
@@ -145,9 +145,7 @@ bool
 instr_is_cbr_arch(instr_t *instr)
 {
     int opc = instr->opcode; /* caller ensures opcode is valid */
-    return (opc == OP_bcond ||
```
[the remark this answers is not in the document]
Comment (derekbruening) [reply core/arch/aarch64/instr.c]: There are special comments that instruct clang-format to ignore that part of the code.

File: .clang-format (hunk 2 above)
Comment (fhahn) [reply .clang-format]: I am using clang-format 6.0

File: .clang-format (hunk 3 above)
Comment (fhahn) [reply .clang-format]: Done

File: .clang-format (hunk 1 above)
Comment (derekbruening) [reply .clang-format]: Right, I guess there's no ideal solution (llvm.org having clang-format as a separate download :)).  I guess we can explore this separately from this PR.

**Comment by fhahn** [comment]
That sounds good! I'll commit the clang-format style file only (plus the comment to avoid reordering the properly aligned condition). If we decide to re-format everything, we can do so in a single commit.

> Have you checked what version is on Travis (Trusty)?
According to https://packages.ubuntu.com/search?keywords=clang-format&searchon=names

* trusty: 3.9
* xenial-updates: 5.0
* bionic: 6.0

File: .clang-format (hunk 1 above)
Comment (fhahn) [reply .clang-format]: It seems like http://apt.llvm.org/ provides APT mirrors for debian and ubuntu, which provide clang-format as a separate package (on X86 at least).

**Pull request merged**